tests/unit/components/sl-date-picker-test.js

```diff
@@ -367,22 +367,48 @@ test( 'updateDateRange() - when date outside endDate range we show placeholder t
     );
 });
 
-test( 'unregisterEvents() - listens to correct event', function( assert ) {
+test( 'changeDate listener is added and removed from the correct namespace', function( assert ) {
     const component = this.subject();
-    const inputElement = this.$( 'input.date-picker' )[ 0 ];
+    const inputElement = this.$( 'input.date-picker' );
+    const jQueryData = Ember.get( Ember.$, '_data' );
+    const hasDatePickerNamespace = () => {
+        let hasNamespace = false;
+
+        assert.equal(
+            Ember.typeOf(
+                Ember.get( jQueryData( inputElement[0], 'events' ), 'changeDate' )
+            ),
+            'array',
+            'Datepicker has changeDate listeners'
+        );
 
-    assert.equal(
-        Ember.typeOf( $._data( inputElement, 'events' ).changeDate ),
-        'array',
-        'Datepicker has a changeDate event listener after render'
+        Ember.get( jQueryData( inputElement[0], 'events' ), 'changeDate' ).every(
+            ( element ) => {
+                if ( element.namespace === "sl-date-picker" ) {
+                    hasNamespace = true;
+                    return false;
+                }
+                return true;
+            }
+        );
+
+        return hasNamespace;
+    };
+
+    assert.ok(
+        hasDatePickerNamespace(),
+        'Datepicker has a changeDate event listener in the correct namespace after render'
     );
 
-    component.trigger( 'willClearRender' );
+    inputElement.on( 'changeDate', function(){} );
```

Comment:
it's to add a changeDate listener not in our namespace, so we can verify that willClearRender does not remove it